tutorials/dapt-curation/code/main.py

```diff
@@ -0,0 +1,236 @@
+    redact_code,
+)
+
+import nemo_curator as nc
+from nemo_curator import ExactDuplicates, Modify, ScoreFilter, Sequential
+from nemo_curator.datasets import DocumentDataset
+from nemo_curator.filters import RepeatingTopNGramsFilter, WordCountFilter
+from nemo_curator.modifiers.pii_modifier import PiiModifier
+from nemo_curator.modifiers.unicode_reformatter import UnicodeReformatter
+from nemo_curator.utils.distributed_utils import get_client
+from nemo_curator.utils.file_utils import (
+    get_all_files_paths_under,
+    separate_by_metadata,
+)
+from nemo_curator.utils.script_utils import ArgumentHelper
+
+SCRIPT_DIR_PATH = os.path.dirname(os.path.abspath(__file__))
+DATA_DIR = os.path.join(SCRIPT_DIR_PATH, "data")
+
+
+def download_sources(
+    wikipedia_limit: Optional[int] = None,
+    github_limit: Optional[int] = None,
+    pdf_limit: Optional[int] = None,
+) -> Tuple[str, str]:
+    """
+    Downloads all the dataset sources and converts them to the JSONL format.
+
+    Returns:
+        tuple: the list of text files and the list of code files.
+    """
+
+    wikipedia_dir = download_wikipedia_sources(
+        "sources/wikipedia_urls.jsonl", limit=wikipedia_limit
+    )
+    github_dir = download_github_sources(
+        "sources/github_repos.jsonl", limit=github_limit
+    )
+    pdf_dir = download_pdf_sources("sources/arxiv_urls.jsonl", limit=pdf_limit)
+
+    wiki_files = get_all_files_paths_under(wikipedia_dir)
+    code_files = get_all_files_paths_under(github_dir)
+    pdf_files = get_all_files_paths_under(pdf_dir)
+
+    text_files = wiki_files + pdf_files
+
+    return text_files, code_files
+
+
+def plot_data(orig_dataset: DocumentDataset, filename: str):
+    """
+    Plot histogram of different file types and corresponding sizes
+
+    Args:
+        dataset (DocumentDataset): Dataset
+        filename (str): Name of the plot to be saved ('sample.png')
+    """
+    # visualize file types and sizes
+    orig_df = orig_dataset.df.compute()
+    orig_df = orig_df.reset_index()
+
+    # Create a histogram for different file types -text
+    fig, ax = plt.subplots(figsize=(10, 6))
+    orig_df.groupby("file_extension")["size_in_bytes"].sum().plot(kind="bar", ax=ax)
+    ax.set_xlabel("file_extension")
+    ax.set_ylabel("size_in_bytes")
+    ax.set_title("File Size Histogram by File Extension")
+
+    # Save the histogram to a file
+    fig.savefig(filename, bbox_inches="tight")
+
+
+def run_curation_pipeline(args: Any, text_files: str, code_files: str) -> None:
+    """
+    Run the curation pipeline on the TinyStories dataset.
+
+    Args:
+        args (Any): Command-line arguments.
+        jsonl_dir (str): Directory path where the JSONL files are stored.
+    """
+    print("Running the curation pipeline...")
+    # Initialize the Dask cluster.
+    client = get_client(**ArgumentHelper.parse_client_args(args))
+
+    # Define data curation steps for text and pdf files
+    curation_steps_text = Sequential(
```

Comment:
(Comment applies to the block below as well)

I think the ordering needs to change a bit:

1. clean and unify
2. dedupe
3. filter

The reason for doing `unify` first is to ensure dedupe operates on unified text and finds (potentially) more duplicates.

Reply:
Corrected it. I assumed after deduplication, filter and clean would work on lesser number of files and hence computation reduces. Makes sense.

Reply:
Yes exactly. Thanks.